db/getItemsOnList.go

```diff
@@ -0,0 +1,39 @@
+package db
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/mount-joy/thelist-lambda/data"
+)
+
+func (d *db) GetItemsOnList(listID *string) (*[]data.Item, error) {
+	input := &dynamodb.QueryInput{
+		ExpressionAttributeValues: map[string]*dynamodb.AttributeValue{
+			":id": {S: listID},
+		},
+		KeyConditionExpression: aws.String("ListId = :id"),
+		TableName:              aws.String(dbTableNameItems),
+	}
+
+	result, err := d.session.Query(input)
+	if err != nil {
+		return nil, err
+	}
+	if result.Items == nil {
+		return nil, nil
+	}
+
+	items := []data.Item{}
+
+	for _, i := range result.Items {
+		item := new(data.Item)
+		err = dynamodbattribute.UnmarshalMap(i, &item)
+		if err != nil {
+			return nil, err
```

Comment:
If we have managed to get through some items before we errored, would there be value in returning the half complete list that was got successfully?

Reply:
I'm not sure there is, especially as it'd be pretty catastrophic if this was failing, right?